### PR TITLE
refactor：plugin adjusts to the left and right sides

### PR DESCRIPTION
**Pull request by STATICHIT**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Background and solution
1.将addons中的setting数组合并到plugins中，并处理项目相关处代码
2.所有带二级面板的插件根据插件所在位置判断显示位置
3.所有插件根据所在位置进行阴影适应

### What is the current behavior?
所有插件在左右侧均可适配

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `align` property in the `PluginSetting` component for enhanced alignment control.
	- Added dynamic styling capabilities for various components based on alignment settings.

- **Improvements**
	- Enhanced layout responsiveness and styling across multiple components by utilizing computed properties for margin and alignment.
	- Streamlined the styling logic to improve readability and maintainability.

- **Bug Fixes**
	- Removed deprecated settings handling in the plugin configuration, simplifying the plugin management process.

### CODE REVIEW

**Review by lizhijie429 — APPROVED** [review]

File: packages/common/component/PluginSetting.vue [hunk 1]
```diff
@@ -133,10 +146,22 @@ export default {
...
+      box-shadow: 6px 0px 3px 0px rgba(0, 0, 0, 0.05);
+      border-right: none;
+    }
+
+    &.shadow-left {
+      box-shadow: -6px 0px 3px 0px rgba(0, 0, 0, 0.05);
+      border-left: none;
+    }
   }
+  // &:not(.second-panel) {
```
Comment (hexqi) [inline packages/common/component/PluginSetting.vue]: 没用到的代码可以删掉

File: packages/common/component/PluginSetting.vue [hunk 2]
```diff
@@ -109,14 +114,22 @@ export default {
...
       return isFullScreen ? '收起' : '全屏查看'
     }
-
+    // 计算属性用于确定阴影类
+    const shadowClass = computed(() => {
+      if (props.align.includes('right')) {
+        return !props.isSecond ? 'shadow-right' : ''
+      } else {
+        return !props.isSecond ? 'shadow-left' : ''
+      }
```
Comment (hexqi) [inline packages/common/component/PluginSetting.vue]: 改成这样更简洁更易理解一点：
```js
if (props.isSecond) return ''
return props.align.includes('right') ? 'shadow-right' : 'shadow-left'
```

File: packages/common/component/PluginSetting.vue [hunk 3]
```diff
@@ -1,6 +1,7 @@
 <template>
   <div
-    :class="['plugin-setting', { 'second-panel': isSecond }, { 'full-screen': state.isFullScreen }]"
+    :style="{ right: align.includes('right') ? 0 : null }"
```
Comment (hexqi) [inline packages/common/component/PluginSetting.vue]: 尽量优先用class而不是内联样式，可以添加一个class，在:class中动态设置

File: packages/common/component/PluginSetting.vue [hunk 4]
```diff
@@ -109,14 +114,22 @@ export default {
...
+      }
+    })
     return {
       state,
       fullScreen,
       getFullScreenLabel,
       IconFullscreen: iconFullscreen(),
       IconMinscreen: iconMinscreen(),
-      IconClose: iconClose()
+      IconClose: iconClose(),
```
Comment (hexqi) [inline packages/common/component/PluginSetting.vue]: icon不在setup中return，写到components里面

File: packages/plugins/bridge/src/BridgeSetting.vue [hunk 5]
```diff
@@ -153,8 +153,14 @@ export default {
...
   },
   setup(props, { emit }) {
-    const { getPluginWidth, PLUGIN_NAME } = useLayout()
-    const leftMargin = computed(() => getPluginWidth(PLUGIN_NAME['Bridge']))
+    const { getPluginWidth, PLUGIN_NAME, getPluginByLayout } = useLayout()
+    const align = computed(() => getPluginByLayout(PLUGIN_NAME['Bridge']))
+    const margin = computed(() => getPluginWidth(PLUGIN_NAME['Bridge']))
+    // 计算样式
+    const computedStyle = computed(() => {
+      return align.value.includes('left') ? { marginLeft: margin.value + 'px' } : { marginRight: margin.value + 'px' }
```
Comment (hexqi) [inline packages/plugins/bridge/src/BridgeSetting.vue]: 可以简化成：`return { [align.value.includes('left') ? 'marginLeft' : 'marginRight']: margin.value + 'px' }`

File: packages/plugins/page/src/PageFolderSetting.vue [hunk 6]
```diff
@@ -82,8 +83,12 @@ export default {
...
     const { pageSettingState, changeTreeData } = usePage()
     const { confirm } = useModal()
-    const { getPluginWidth, PLUGIN_NAME } = useLayout()
-    const leftMargin = computed(() => getPluginWidth(PLUGIN_NAME['AppManage']))
+    const { getPluginWidth, PLUGIN_NAME, getPluginByLayout } = useLayout()
+    const align = computed(() => getPluginByLayout(PLUGIN_NAME['AppManage']))
+    const margin = computed(() => getPluginWidth(PLUGIN_NAME['AppManage']))
+    const computedStyle = computed(() => {
+      return align.value.includes('left') ? { marginLeft: margin.value + 'px' } : { marginRight: margin.value + 'px' }
+    })
```
Comment (hexqi) [inline packages/plugins/page/src/PageFolderSetting.vue:91]: 每个插件都要在内部去做这段代码的适配吗，看下有没有更好的方式，比如有没有可能在公共组件上实现不用在具体的组件里面去适配

Comment (STATICHIT) [reply packages/plugins/page/src/PageFolderSetting.vue:91]: 这些是插件的二级页面，二级页面的实现方式本身也不太一样，使用的组件也不仅只在二级页面中使用到，所以得在具体的二级页面中适配。并不是每个插件都需要做适配，只是各别二级页面。

**Review by hexqi — APPROVED** [review]

**Review by hexqi — CHANGES_REQUESTED** [review]
@STATICHIT commit没有签名，需要改下

**Review by hexqi — APPROVED** [review]

**Pull request merged**